### ISSUE TITLE
Documentation for Portal API

### ISSUE THREAD

**Issue by lyptt** [lonelycode/tyk#64]
Is there any chance of releasing API documentation for the Portal? We're using our own self-hosted portal which we'll need to hook into Tyk. Although there seems to be documentation for the main Tyk API, anything to do with the Dashboard seems to be missing.

The kind of APIs I'm interested in are those about key management, although it would be great to get the API catalogue info too if there's an API for that.

Here's some of the ones I've observed being used so far:

```POST /portal/member/apis/{{API ID}}/request``` for requesting an API key
```GET /api/portal/developers/{{DEV ID}}``` for retrieving developer information
```DELETE /api/portal/developers/key/{{KEY}}``` for deleting an API key

Basically if it would be possible to get some kind of API for managing end-users, that would be brilliant. Alternatively if there are any plans to open source the Dashboard, we could work on this ourselves and contribute it back.

**Comment by lonelycode** [comment]
Hi,

Yes there is afull portal API - the dashboard is essentially a client of it (it's an angular app). We may publish the documentation on that, it depends on demand, there's quite a lot of documentation :-)

Some of the dashboard API is documented under "Advanced Management API" on our site so you can do key management through this quite easily, but not anything to do witht he portal.

We're unlikely to open source the dashboard any time soon, we are looking for partners to bring on board that can work with us on the repo, but there would need to be some serious interest in wanting to get involved. the details for that haven;t been ironed out yet though.

Basically though, if you can do it in the dashboard interface there's an API for it :-)

**Comment by lyptt** [comment]
That's understandable. Documentation for the portal-specific portions would be incredibly useful. We can get around our issue by just creating and managing Tyk users instead of portal developers, but it would be nice to have a clean separation between organization users and third party users.

**Comment by lonelycode** [comment]
BTW Tyk users are basically admins, so I wouldn't use those for your developers :-)

The need is understandable - and it'll be done, eventually... it's quite a large system now.
